src/compiler/crystal/tools/print_hierarchy.cr

```diff
@@ -95,6 +94,41 @@ module Crystal
       false
     end
 
+    def must_print?(type : NonGenericClassType)
+      !(@exp && !@targets.includes?(type))
```

Comment:
IMO the De Morgan equivalent is a bit easier to follow than the double negation:
```suggestion
      !@exp || @targets.includes?(type))
```
Ditto below.